Design Updates, Error Messages, and Form Field Adjustments

nadezdatsygankova opened the pull request:
This PR enhances the sign-up form based on the latest Figma design specifications. 
Design Update

Figma Alignment: Updated the sign-up form to match the latest Figma design version.
Styling Enhancements: Improved the visual styling for inputs, buttons, and labels.

Form Field Modifications

Added Fields: Introduced input fields for First Name and Last Name.
Removed Field: Eliminated the Password Confirmation field to streamline the user registration process.

Error Message Implementation

Inline Error Messages: Added inline error messages for invalid inputs, providing immediate feedback to users.
Visual Feedback: Incorporated visual feedback for errors, such as red borders and error icons next to problematic fields.

Backend Update:

User Model: Updated the user model to include firstName and lastName fields.

Screenshots:

Figma Design:
![Screenshot 2024-06-14 at 10 27 39 PM](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/47821694/1f2c2f29-e4de-4f8d-abf3-c52730e86f92)


Before Update

![Screenshot 2024-06-14 at 10 13 04 PM](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/47821694/dd5fd3bf-c395-42a7-abfc-71d3a6859304)

After Update 
[
![Screenshot 2024-06-14 at 9 57 15 PM](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/47821694/8a888e05-704a-468f-a676-2d77f3711716)
](url)

Testing Instructions
Navigate to the Sign-Up Page: Confirm the form layout and styling match the Figma design.
Input Validation: Test the new fields (First Name, Last Name) for proper validation.
Error Handling: Verify error messages appear correctly when invalid data is entered.
Form Submission: Ensure the form is submitted correctly without the password confirmation field. 

Note. 
Please, during testing, disregard the console error related to App.jsx:33. This issue is unrelated to the sign-up form and is being addressed separately.

VrushaliHaldankar-01 committed:
added registration form design

VrushaliHaldankar-01 committed:
changes in register form

VrushaliHaldankar-01 committed:
changed files register

VrushaliHaldankar-01 committed:
added form component

mabelzhou committed:
added registration name requirement to backend

mabelzhou committed:
fixed typo

nadezdatsygankova committed:
fixed errors

nadezdatsygankova committed:
fixed color

VrushaliHaldankar-01 committed:
changes made in register form

VrushaliHaldankar-01 committed:
Merge branch 'beavers-sign-up-new-version' of https://github.com/JULIERAJ/KIDS-FIRST-v2 into beavers-sign-up-new-version

nadezdatsygankova committed:
merge from main into beavers

nadezdatsygankova committed:
changed focus text

VrushaliHaldankar-01 committed:
changes in sign up design

VrushaliHaldankar-01 committed:
Merge branch 'beavers-sign-up-new-version' of https://github.com/JULIERAJ/KIDS-FIRST-v2 into beavers-sign-up-new-version

VrushaliHaldankar-01 committed:
changes in sign up design

nadezdatsygankova committed:
changed bgc

KatePolyakov committed:
changed styles according design

KatePolyakov committed:
changed css rules for inputs

AnnaSolovykh committed:
fix focus on password input, refactor css module

AnnaSolovykh committed:
add input border and shadow

AnnaSolovykh committed:
improve input focus

nadezdatsygankova committed:
fix(auth):resolve label error in the console

AnnaSolovykh committed:
resolve merge conflict

AnnaSolovykh committed:
Merge pull request #101 from JULIERAJ/input-focus-styling

AnnaSolovykh committed:
Merge pull request #91 from JULIERAJ/beavers-sign-up-new-version--test

AnnaSolovykh commented on frontend/src/pages/Register/Register.jsx:
```diff
@@ -1,18 +1,21 @@
 import { register } from '../../api';
-import FatherSonBlock from '../../components/FatherSonBlock';
+// import FatherSonBlock from '../../components/FatherSonBlock';
 import Header from '../../components/Header';
-import TextLink from '../../components/TextLink';
+// import TextLink from '../../components/TextLink';
```
Will we need commented out code in the future?

VrushaliHaldankar01 replied:
Hii all the commented lines from the code has been removed.

AnnaSolovykh commented on frontend/src/pages/Register/Register.module.css:
```diff
@@ -1,12 +1,166 @@
+  letter-spacing: 2%;
+  color: #081821;
+}
+.EmailLabel,
```
Why is PascalCase used for these classnames? Is it occasionally or on purpose? If occasionally, let's follow the common pattern and use camelCase.

VrushaliHaldankar01 replied:
The class name has been changed

AnnaSolovykh commented on frontend/src/pages/Register/Register.module.css:
```diff
@@ -1,12 +1,166 @@
+  height: 56px;
+  /* Add other button styles here */
+}
+.SocialButton {
```
Why is PascalCase used for these classnames? Is it occasionally or on purpose? If occasionally, let's follow the common pattern and use camelCase.

AnnaSolovykh commented on frontend/src/pages/Register/Register.module.css:
```diff
@@ -1,12 +1,166 @@
\ No newline at end of file
+  color: aliceblue;
+}
+/* In Register.module.css or your preferred CSS file */
```
This is a bit strange comment :) Comments should be added to explain some difficult patterns to the project's new-comers. 
I don't think we need this comment :)

AnnaSolovykh commented on frontend/src/pages/Register/Register.module.css:
```diff
@@ -1,12 +1,166 @@
+
+.orDivider {
+  margin-top: 24px;
+  text-align: center; /* Align the text center */
+  margin: 20px 0; /* Adjust the margin as needed */
+}
+
+.dashLine {
+  position: relative; /* Set position to relative */
+  color: #454848; /* Set the color */
+}
+
+.dashLine:before,
+.dashLine:after {
+  content: '';
+  position: absolute; /* Set position to absolute */
+  top: 50%; /* Set top position to center vertically */
+  width: 200px; /* Set the width */
+  height: 1px; /* Set the height */
+  background-color: #454848; /* Set the color */
+}
+
+.dashLine:before {
+  left: -220px; /* Adjust the position of the first dash line */
+}
+
+.dashLine:after {
+  right: -220px; /* Adjust the position of the second dash line */
```
Is making these comments justified? Seems like they are explaining basic stuff.

AnnaSolovykh commented on frontend/src/pages/Register/Register.module.css:
```diff
@@ -1,12 +1,166 @@
+}
+.loginLink {
+  cursor: pointer;
+  color: #f08324; /* Set color to white */
+  text-decoration: underline; /* Add underline */
```
The same. The color isn't even white. 
<img width="411" alt="Screenshot 2024-06-17 at 11 33 59" src="https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/114008959/ec85fa71-cd6d-47f8-8dc1-1327eb41f723">

AnnaSolovykh reviewed (changes requested):
Well done! Everything looks really nice, good job! 
Though there are a few minor issues to look into. 
Also Log In button under the register form does not work.
<img width="500" alt="Screenshot 2024-06-17 at 11 35 45" src="https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/114008959/22d9f41e-43e7-4ffe-916d-db313927f08f">

VrushaliHaldankar01 committed:
Resolved css errors in Register module css file

AnnaSolovykh commented:
@VrushaliHaldankar01 thanks, Vrushali! Looks good to me!

We also need to fix log in link on the register page. Could you be so kind to fix it too?

AnnaSolovykh commented:
@nadezdatsygankova @KatePolyakov @VrushaliHaldankar-01 look, also I don't see this error in the dashboard branch - I guess it should be fixed before merging. Who can take it?
<img width="353" alt="Screenshot 2024-06-17 at 20 12 05" src="https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/114008959/49a62d1a-e795-4e53-a758-8572ff785f75">

nadezdatsygankova committed:
fix: resolve login navigation button link

nadezdatsygankova committed:
fix: removed commented lines

nadezdatsygankova commented:
> Well done! Everything looks really nice, good job! Though there are a few minor issues to look into. Also Log In button under the register form does not work. <img alt="Screenshot 2024-06-17 at 11 35 45" width="500" src="https://private-user-images.githubusercontent.com/114008959/340363705-22d9f41e-43e7-4ffe-916d-db313927f08f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NzkzOTIsIm5iZiI6MTcxODY3OTA5MiwicGF0aCI6Ii8xMTQwMDg5NTkvMzQwMzYzNzA1LTIyZDlmNDFlLTQzZTctNGZmZS05MTZkLWRiMzEzOTI3ZjA4Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOFQwMjUxMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kOTViZDVhYWFjMTkyY2ExYzMyNzZjNzljYmRjMmNhMmY3OTNjZmQ5MzQ0ZTFjMzgzNTg3M2UxOGNkZjJiNGM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NiMWnChiQjkhkLmLwA-QUn9tHW4HJzyWS2vBMqe_pE0">

Thank you for catching these errors. 
I've fixed the link in the 'login in' button and removed commented lines.

nadezdatsygankova commented:
> @VrushaliHaldankar01 thanks, Vrushali! Looks good to me!
> 
> We also need to fix log in link on the register page. Could you be so kind to fix it too?

I've fixed log in link

nadezdatsygankova committed:
fix: resolve console error

nadezdatsygankova commented:
> @nadezdatsygankova @KatePolyakov @VrushaliHaldankar-01 look, also I don't see this error in the dashboard branch - I guess it should be fixed before merging. Who can take it? <img alt="Screenshot 2024-06-17 at 20 12 05" width="353" src="https://private-user-images.githubusercontent.com/114008959/340489922-49a62d1a-e795-4e53-a758-8572ff785f75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NzkzOTEsIm5iZiI6MTcxODY3OTA5MSwicGF0aCI6Ii8xMTQwMDg5NTkvMzQwNDg5OTIyLTQ5YTYyZDFhLWU3OTUtNGU1My1hNzU4LTg1NzJmZjc4NWY3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOFQwMjUxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZTVlODNhNmYzZjhiNjkzZTU3OWY0NTdiZmQxNTZjMGFhNWZlZDY2MTFmZjExODY3MTRhNjI4YzM4ODdmODk5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QVOF8iKLVjjePcUvqfpzkywkvsTRIDyzwclO5hwpjvY">

I've fixed it. Thank you

KatePolyakov reviewed (approved):
Well Done Ladies

S-chahine reviewed (commented):
Great job, ladies! I have tested the form submission process, and everything is working fine overall. However, I encountered a few errors that I'm not sure about:

Sign-Up Button Error:
When I click the Sign-Up button, I get the following error:
![image_720 (1)](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/86498834/e30249f5-1d1d-4842-b18b-7d3d4a7b410a)


Page Load Error:
Once the page is loaded, this error appears in the console:


![image_480](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/86498834/5e82bf49-2251-4474-a7fb-cc173b753d2c)

Login Button Error:
When I click the Login button, this error message appears:
![image_720 (1)](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/86498834/fcf5243d-cb9d-4dc3-b444-710962355c2b)

Could you please take a look at these issues? Thanks!

AnnaSolovykh commented:
> Great job, ladies! I have tested the form submission process, and everything is working fine overall. However, I encountered a few errors that I'm not sure about:
> 
> Sign-Up Button Error: When I click the Sign-Up button, I get the following error: ![image_720 (1)](https://private-user-images.githubusercontent.com/86498834/340876546-e30249f5-1d1d-4842-b18b-7d3d4a7b410a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3NjQ3NDUsIm5iZiI6MTcxODc2NDQ0NSwicGF0aCI6Ii84NjQ5ODgzNC8zNDA4NzY1NDYtZTMwMjQ5ZjUtMWQxZC00ODQyLWIxOGItN2QzZDRhN2I0MTBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDAyMzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwZmU3MmZmOTA2NDFhOTM0N2NhYzAzOWExYjhkMTY4YjQ3MWVkNjMwMjU5Y2NkNTFlMzc2NmVmMTM2MzRiYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.P46sMKBmZLbTByTRAkn4xvRpL8P9-cmAuectohDh7Mc)
> 
> Page Load Error: Once the page is loaded, this error appears in the console:
> 
> ![image_480](https://private-user-images.githubusercontent.com/86498834/340876592-5e82bf49-2251-4474-a7fb-cc173b753d2c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3NjQ3NDUsIm5iZiI6MTcxODc2NDQ0NSwicGF0aCI6Ii84NjQ5ODgzNC8zNDA4NzY1OTItNWU4MmJmNDktMjI1MS00NDc0LWE3ZmItY2MxNzNiNzUzZDJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDAyMzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3MWFhZDQ1ZjViYWE0YTAzNjgwZjQxYmY2ZjQ2YmQxYTY3ZmNmYWUwNDY3YzNiZTNkYTgzMjJiZmU1ZTY3MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qPy2GJDFAP4PIO2cIhMgq5lDOFpacbMbajmQmeqchIs)
> 
> Login Button Error: When I click the Login button, this error message appears: ![image_720 (1)](https://private-user-images.githubusercontent.com/86498834/340876617-fcf5243d-cb9d-4dc3-b444-710962355c2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3NjQ3NDUsIm5iZiI6MTcxODc2NDQ0NSwicGF0aCI6Ii84NjQ5ODgzNC8zNDA4NzY2MTctZmNmNTI0M2QtY2I5ZC00ZGMzLWI0NDQtNzEwOTYyMzU1YzJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDAyMzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MzU3ZDRlYTUyMWVhOWMyMTJkOTBiNDYxOTg3YTA5ZDdiOWRkZDMyMjVjNmY3YmI2MDNjNDc0N2MyMGEwYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kNmEpLBT_eRxPek6P9GkPR_7CmSXwTWk0L5GknUzCuY)
> 
> Could you please take a look at these issues? Thanks!

I have put these errors in the separate ticket  - https://github.com/users/JULIERAJ/projects/2/views/1?pane=issue&itemId=67482131  We will need a volunteer tomorrow to fix them :)

Skyline857 reviewed (approved):
Amazing job, ladies! I've tested the submission form. Everything looks great overall! However, when I try to click on Sign- Up button, I see these type of errors in my console

![registertest](https://github.com/JULIERAJ/KIDS-FIRST-v2/assets/124208414/82d33b08-3a04-4ea3-893c-1943b837536d)

AnnaSolovykh commented:
> Amazing job, ladies! I've tested the submission form. Everything looks great overall! However, when I try to click on Sign- Up button, I see these type of errors in my console
> 
> ![registertest](https://private-user-images.githubusercontent.com/124208414/341134062-82d33b08-3a04-4ea3-893c-1943b837536d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MTM1MDgsIm5iZiI6MTcxODgxMzIwOCwicGF0aCI6Ii8xMjQyMDg0MTQvMzQxMTM0MDYyLTgyZDMzYjA4LTNhMDQtNGVhMy04OTNjLTE5NDNiODM3NTM2ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOVQxNjA2NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZjNkMzE4YzE3ZjE4YmUwZjdlMDc4YzU4Y2Q4MDdhYTYwYmY0MzM1ZjVjZjM0ZjQ3MWQ2YTgzMzc1Y2YxNTE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WT732wTsbJrl0Z2AhCHP0oPM1OSQu5PNZHA1BeEGiMM)

This is fine - we are not connected to the api yet.
